File: delete_all_tasks.py
#!/usr/bin/env python3
"""
Script để xóa toàn bộ tasks trong Firebase Firestore
"""

import firebase_admin
from firebase_admin import credentials, firestore
import os

def delete_all_tasks():
    """Xóa toàn bộ tasks trong collection 'tasks'"""
    
    try:
        # Khởi tạo Firebase Admin SDK
        if not firebase_admin._apps:
            # Sử dụng service account key nếu có
            if os.path.exists('serviceAccountKey.json'):
                cred = credentials.Certificate('serviceAccountKey.json')
                firebase_admin.initialize_app(cred)
                print("✅ Đã khởi tạo Firebase với service account key")
            else:
                # Sử dụng Application Default Credentials
                firebase_admin.initialize_app()
                print("✅ Đã khởi tạo Firebase với Application Default Credentials")
        
        # Lấy Firestore client
        db = firestore.client()
        print("✅ Đã kết nối Firestore")
        
        # Lấy collection 'tasks'
        tasks_ref = db.collection('tasks')
        
        # Lấy tất cả documents
        docs = tasks_ref.stream()
        task_count = 0
        
        print("🔍 Đang tìm tasks để xóa...")
        
        # Xóa từng document
        for doc in docs:
            print(f"🗑️ Đang xóa task: {doc.id}")
            print(f"   Data: {doc.to_dict()}")
            doc.reference.delete()
            task_count += 1
        
        print(f"✅ Đã xóa thành công {task_count} tasks!")
        
        # Kiểm tra lại
        remaining_docs = tasks_ref.stream()
        remaining_count = sum(1 for _ in remaining_docs)
        
        if remaining_count == 0:
            print("✅ Collection 'tasks' đã trống!")
        else:
            print(f"⚠️ Vẫn còn {remaining_count} tasks (có thể do lỗi)")
            
    except Exception as e:
        print(f"❌ Lỗi: {e}")
        return False
    
    return True

def main():
    print("🚀 Bắt đầu xóa toàn bộ tasks...")
    print("=" * 50)
    
    success = delete_all_tasks()
    
    print("=" * 50)
    if success:
        print("✅ Hoàn thành!")
    else:
        print("❌ Thất bại!")

if __name__ == "__main__":
    main() 

File: delete_all_tasks.js
const admin = require('firebase-admin');

// Khởi tạo Firebase Admin SDK
if (!admin.apps.length) {
  try {
    // Thử sử dụng service account key
    const serviceAccount = require('./serviceAccountKey.json');
    admin.initializeApp({
      credential: admin.credential.cert(serviceAccount)
    });
    console.log('✅ Đã khởi tạo Firebase với service account key');
  } catch (error) {
    // Sử dụng Application Default Credentials
    admin.initializeApp();
    console.log('✅ Đã khởi tạo Firebase với Application Default Credentials');
  }
}

const db = admin.firestore();

async function deleteAllTasks() {
  try {
    console.log('🚀 Bắt đầu xóa toàn bộ tasks...');
    console.log('='.repeat(50));
    
    // Lấy collection 'tasks'
    const tasksRef = db.collection('tasks');
    
    // Lấy tất cả documents
    const snapshot = await tasksRef.get();
    
    if (snapshot.empty) {
      console.log('✅ Không có tasks nào để xóa!');
      return;
    }
    
    console.log(`🔍 Tìm thấy ${snapshot.size} tasks để xóa...`);
    
    let deletedCount = 0;
    
    // Xóa từng document
    for (const doc of snapshot.docs) {
      console.log(`🗑️ Đang xóa task: ${doc.id}`);
      console.log(`   Data:`, doc.data());
      
      await doc.ref.delete();
      deletedCount++;
    }
    
    console.log(`✅ Đã xóa thành công ${deletedCount} tasks!`);
    
    // Kiểm tra lại
    const remainingSnapshot = await tasksRef.get();
    
    if (remainingSnapshot.empty) {
      console.log('✅ Collection "tasks" đã trống!');
    } else {
      console.log(`⚠️ Vẫn còn ${remainingSnapshot.size} tasks (có thể do lỗi)`);
    }
    
  } catch (error) {
    console.error('❌ Lỗi:', error);
  }
}

// Chạy script
deleteAllTasks().then(() => {
  console.log('='.repeat(50));
  console.log('✅ Hoàn thành!');
  process.exit(0);
}).catch((error) => {
  console.error('❌ Thất bại:', error);
  process.exit(1);
}); 